### ISSUE TITLE
Repository layout: adopt a standard master branch with all supported versions

### ISSUE THREAD

**Issue by jouvin** [quattor/template-library-grid#248]
For reasons I forgot, when we created this repo at SVN -> Git migration time, we opted for a branch per middleware version and no master branch. It turned out not to be a great idea: it makes everything more complicated, including:

- A release must tag several branches
- Fixing a problem affecting several middleware versions requires several PRs
- get-template-library must do complicated things to select all the branches to checkout
- Obsoleting a branch requires renaming it with a `.obsolete` suffix rather than removing the corresponding set of templates in the `master` branch

My proposal is to move to a standard structure with a `master` branch and have a first level directory representing the middleware version the templates must be used for, e.g. `umd-4`, `umd-5 `... and to adapt the release tools and `get-template-library` to deal with this new/simpler structure.

Does anybody disagree? It should be transparent to `template-library-grid` users, in fact it should make their life easier. If there is an agreement, do you want to do it in the upcoming release, 24.10? Adding explicitely @jrha and @Pansanel .

Proposed plan includes:
- [x] Create new `main` branch from `umd-4` branch
- [x] Adapt release tools to the new (simpler) layout
- [x] Update the repository structure description in `get-library-template`

Ideally, should be done before adding `umd-5` templates to simplify the operation and avoid the need to "merge" 2 branches with their history.

**Comment by jrha** [comment]
Duplicate of https://github.com/quattor/release/issues/309

**Comment by jrha** [comment]
We are definitely not doing it this release though, it will break too many things in the release process.

**Comment by jouvin** [comment]
No I don't think it duplicates https://github.com/quattor/release/issues/309 that I forgot and tend to disagree with... Ok, I'll try to make the change once 24.10 is released.

**Comment by jrha** [comment]
https://github.com/quattor/release/pull/370/files
